### DISCUSSION TITLE
Engineering

### DISCUSSION THREAD

**Discussion by colinkiama** [nahuelwexd/vala-website#11] [General]
My current plan is to switch to [eleventy](https://github.com/11ty/eleventy) as it’s lightweight and allows us to be flexible about how the site is built, making it easier for more people to contribute.

Ultimately, I would like the site to be generated using some sort of Vala static site generator (or something else that doesn’t depend on Node.js). It’s an opportunity to make build times faster and prerequisites simpler (the generator would be in one binary instead of having to install an entire runtime).

Feel free to give your feedback about this below.

**Comment by colinkiama** [comment]
I ended up switching to [Zola](https://www.getzola.org) instead. Just installing the dependencies alone for eleventy resulted in a 60MB `node_modules` folder.

Zola meets the requirement of having simple prerequisites (it's a single binary), fast build times and also it doesn't require you to install Node.js or any other runtime.

The entire repo folder is around 2MB right now (and that's with some pages and assets that have been added too).

I think the much faster onboarding, iteration and deployment experiences make this change worth it.

**Comment by colinkiama** [comment]
Well I've run into an issue with Zola. Internally it uses libsass however, libsass doesn't support the [`@use`](https://sass-lang.com/documentation/at-rules/use) rule. [They do have plans to solve this issue in the future though.](https://github.com/getzola/zola/issues/1535)

As a workaround, [dart-sass](https://github.com/sass/dart-sass) can be added as a dependency. People will need to install both dart-sass and Zola in order to build project locally and some build scripts will need to be created.

Regarding deployment, the current GitHub action used will need to be modified to install a standalone copy of dart-sass first before proceeding to build the site but that can be dealt with later.

If you do have any thoughts about it, please feel free to reply.

**Reply by colinkiama** [reply]
It turns out, this will end up getting to complicated. Just going to continue the work that was being done before I took over since that was working fine. (Yes, I'm switching back to the Next.js setup lol)

**Reply by nyx-lyb3ra** [reply]
Is sass really necessary though? The current version was originally developed using Tailwind

**Reply by colinkiama** [reply]
Sass made sense at the time since the plan was to not use Node.js and npm in the project at all however, we’re using them pretty heavily now 😅.

Tailwind’s integration with NextJS looks really simple too. I’ll switch to it and see how it goes.

**Here are some very strong arguments around it for anybody interested**:
For Tailwind: https://frontstuff.io/in-defense-of-utility-first-css
Against Tailwind (language warning): https://medium.com/codex/why-presentational-classes-for-html-css-are-ignorant-garbage-bcfdb02ec397

**Reply by nyx-lyb3ra** [reply]
I used Tailwind at first as I find it super great for fast iteration over prototypes. The plan was then to switch to Sass and have something decent.

However, since you had problems with Sass in Zola, you could see to use Tailwind with it, although I have already seen and it is the same as if you had used dart-sass

In any case: if you want to use Tailwind, be aware of the mess it causes in the HTML. As I said, it is very useful for fast forwarding and prototyping, but when it's time to clean up the website, I would switch to Sass

**Reply by pervoj** [reply]
I'm a bit against Tailwind. I used that for a while and it is just like you are writing all css but in classes. You also repeat the same thing a lot. For prototyping - yes, but for production it is really messy.

About Sass: we can use Gulp to build SCSS and then run Zola from it. There is also an option to use pure CSS.

I'm still in favor of using Zola. The fact that it takes 1 minute and 30 seconds to clone the repository, install Zola and start the server got to me.

**Reply by colinkiama** [reply]
I'm tired. I'll go back to using Zola after all and work around the lack of `@use` rules with `@import`.

> I used Tailwind at first as I find it super great for fast iteration over prototypes

Browsers themselves have gotten really good for prototyping these days 😯. See: https://developer.chrome.com/docs/devtools/workspaces/

I won't be making any more changes regarding SSGs and styling frameworks.

**Reply by nyx-lyb3ra** [reply]
> See: https://developer.chrome.com/docs/devtools/workspaces/

Doesn't seem to be that comfy for me, but it's fine :)
